src/fields/point_sets.cc

```diff
@@ -0,0 +1,79 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *
+ * @file    point_sets.cc
+ * @brief
+ * @author  David Flanderka
+ */
+
+#include <armadillo>
+#include "fields/point_sets.hh"
+#include "fields/composed_quadrature.hh"
+#include "mesh/side_impl.hh"
+#include "mesh/sides.h"
+
+
+/******************************************************************************
+ * Implementation of BulkSubQuad methods.
+ */
+
+template <unsigned int dim>
+Range< PointAccessor<dim> > BulkSubQuad<dim>::points() const {
+	auto bgn_it = make_iter<PointAccessor<dim>>( PointAccessor<dim>(c_quad_, this->point_indices_[0]) );
+	auto end_it = make_iter<PointAccessor<dim>>( PointAccessor<dim>(c_quad_, this->point_indices_[1]) );
+	return Range<PointAccessor<dim>>(bgn_it, end_it);
+}
+
+
+/******************************************************************************
+ * Implementation of SideSubQuad methods.
+ */
+
+template <unsigned int dim>
+Range< PointAccessor<dim> > SideSubQuad<dim>::points(const Side &side) const {
```

Comment:
The returned range of points must take into account the side permutation. Here we assume always permutation 0.